mmcv/ops/csrc/pytorch/contour_expand.cpp

```diff
@@ -0,0 +1,107 @@
+// It is modified from https://github.com/whai362/PSENet
+#include <iostream>
+#include <queue>
+
+#include "pytorch_cpp_helper.hpp"
+
+using namespace std;
+
+class Point2d {
+ public:
+  int x;
+  int y;
+
+  Point2d() : x(0), y(0) {}
+  Point2d(int _x, int _y) : x(_x), y(_y) {}
+};
+
+void growing_text_line(const uint8_t *data, IntArrayRef data_shape,
+                       const int *label_map,
+
+                       IntArrayRef label_shape, int &label_num, int &min_area,
+                       vector<vector<int>> &text_line) {
+  std::vector<int> area(label_num + 1);
+  for (int x = 0; x < label_shape[0]; ++x) {
+    for (int y = 0; y < label_shape[1]; ++y) {
+      int label = label_map[x * label_shape[1] + y];
+      if (label == 0) continue;
+      area[label] += 1;
+    }
+  }
+
+  queue<Point2d> queue, next_queue;
+  for (int x = 0; x < label_shape[0]; ++x) {
+    vector<int> row(label_shape[1]);
```

Comment:
`label_shape[1]` is used multiple times. Maybe we can give it a name `H` in the begining of the function

Reply:
I will rename it